chromadb/api/shared_system_client.py

```diff
@@ -91,3 +92,58 @@ def _system(self) -> System:
     def _submit_client_start_event(self) -> None:
         telemetry_client = self._system.instance(ProductTelemetryClient)
         telemetry_client.capture(ClientStartEvent())
+
+    @staticmethod
+    def _singleton_tenant_database_if_applicable(
```

Comment:
Nit - should this just take UserIdentity?

Reply:
Would it be cleaner to move this logic into auth?

Reply:
`Client`/`AsyncClient` don't inherently have access to auth, and I think it's cleaner to keep it that way. Slight preference to keep in `SharedSystemClient`, as they are shared client methods.